File: problems/100/solution.py
class Solution:
    def removeDuplicates(self, head):
        if not head or not head.next:
            return head

        current = head
        nextNode = current.next

        duplicateFound = False
        while nextNode and nextNode.val == current.val:
            duplicateFound = True
            nextNode = nextNode.next;

        if duplicateFound:
            current = self.removeDuplicates(nextNode)
        else:
            current.next = self.removeDuplicates(nextNode)

        return current

File: problems/100/solution.cpp
ListNode* Solution::removeDuplicates(ListNode* head) {
	if (!head || !head->next) {
		return head;
	}
	ListNode* current = head;
	ListNode* next = current->next;

	int dup = 0;
	while (next && next->val == current->val) {
		dup = 1;
		next = next->next;
	}

	if (dup == 1) {
		return removeDuplicates(next);
	} else {
		current->next = removeDuplicates(next);
		return current;
	}
}
